spec/API.test.js

```diff
@@ -0,0 +1,44 @@
+const request = require('supertest');
+
+const app = require('../server/app.js');
+
+require('babel-polyfill');
+
+let server, agent;
+
+beforeEach((done) => {
```

Comment:
Take a look at [this](https://www.albertgao.xyz/2017/05/24/how-to-test-expressjs-with-jest-and-supertest/), I used it to test my server and I think it's better than actually starting it. I feel like it might have something to do with why circleci is failing but I'm not sure.

EDIT: I just noticed you are using supertest, I don't think you have to actually call app.listen to make it work. Also I looked at your error message from circleci and the port it mentions is the mongo port, which makes me thing you're trying to start mongo when you don't need to.